### ISSUE TITLE
Floating point exception (core dumped) on torchaudio.load

### ISSUE THREAD

**Issue by Onyoursix** [pytorch/audio#2870]
### 🐛 Describe the bug

Python is crashing when I load certain files with torchaudio. It seems to be fairly random and I'm not sure what is causing the issue. Typically, I download the files and load it from disk, however, the same issue happens with the same files when streamed. Here is a example of a file that actively crashes python. 

I can download the file and it seems to be a working/playable file so I have no clue what's going on. There are no error messages beyond "Floating point exception (core dumped)"

```
import requests
import torchaudio

SAMPLE_WAV_SPEECH_URL = "https://www.utah.gov/pmn/files/877489.WAV"

with requests.get(SAMPLE_WAV_SPEECH_URL, stream=True) as response:
  wav, sr = torchaudio.load(response.raw)

```

### Versions

Collecting environment information...
PyTorch version: 1.13.0+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

Python version: 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-52-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 3060 Ti
GPU 1: NVIDIA GeForce RTX 3080 Ti

Nvidia driver version: 510.85.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.4
[pip3] pytorch-lightning==1.6.5
[pip3] pytorch-metric-learning==1.6.2
[pip3] torch==1.13.0
[pip3] torch-audiomentations==0.11.0
[pip3] torch-pitch-shift==1.2.2
[pip3] torchaudio==0.13.0
[pip3] torchmetrics==0.10.2
[pip3] torchvision==0.14.0
[conda] Could not collect

**Comment by carolineechen** [comment]
When I try rerunning the script with `stream=False` to debug, I end up with the following error, indicating that ffmpeg is unable to pick up on the format of the file

**Comment by carolineechen** [comment]
``

**Comment by mthrok** [comment]
Floating point exception suggests there was zero-division.

Uploading the file to metadata2go.com give [the following result](https://www.metadata2go.com/result/9fe7f23c-cfef-4905-8785-177a93038b2a).

```
File Name: 877489.WAV
File Size: 7.7 MB
File Type: WAV
File Type Extension: wav
Mime Type: audio/x-wav
Encoding: Intel IMA/DVI-ADPCM
Num Channels: 2
Sample Rate: 48000
Avg Bytes Per Sec: 48000
Bits Per Sample: 4
Duration: 0:02:40
Raw Header:
    52 49 46 46
    F8 89 73 0B
    57 41 56 45
    66 6D 74 20
    E4 01 00 00
    11 00
    02 00
    80 BB 00 00
    80 BB 00 00
    00 08
    04 00
    02 00 F9 07
    00 00 00 00
    ^^^^^^^^^^^
    00 00 00 00 ...
```

Two noteworthy points here
 - the encoding is ADPCM, not the regular PCM. 
 - 41-44 bytes of the raw header are all zero, according to the [WAV format specification](https://isip.piconepress.com/projects/speech/software/tutorials/production/fundamentals/v1.0/section_02/s02_01_p05.html#:~:text=The%20WAV%20audio%20format%20was,quality%2016%2Dbit%20audio%20format.), it is supposed to have the size of the data section.

My hypothesis for the error is that this is what is causing zero-division.

---

Now, I do not know a thing about ADPCM format, but it's highly likely that `libsox` (the library backing the `torchaudio.load` function) cannot handle this well.

In my environment, (M1 mac, breading edge torchaudio) the snippet does not fail, but instead it returns 0-length tensor.

```
torch.Size([2, 0]) 48000
```

`ffprobe` reports 01:06:42.52 while `soxi` reports the duration to be 1:06:28.84

```
$ ffprobe 877489.WAV
Input #0, wav, from '877489.WAV':
  Duration: 01:06:42.52, bitrate: 384 kb/s
    Stream #0:0: Audio: adpcm_ima_wav ([17][0][0][0] / 0x0011), 48000 Hz, 2 channels, s16p, 384 kb/s
```

```
$ soxi 877489.WAV

Input File     : '877489.WAV'
Channels       : 2
Sample Rate    : 48000
Precision      : 13-bit
Duration       : 01:06:28.84 = 191464169 samples ~ 299163 CDDA sectors
File Size      : 192M
Bit Rate       : 385k
Sample Encoding: 4-bit IMA ADPCM
```

---

As a workaround, since you are using torchaudio 0.13, I suggest to use `torchaudio.io.StreamReader`, which uses FFmpeg behind the scene and it seems to work fine.

```python
def load(src):
    reader = torchaudio.io.StreamReader(src)
    metadata = reader.get_src_stream_info(reader.default_audio_stream)
    reader.add_basic_audio_stream(-1)
    reader.process_all_packets()
    waveform,  = reader.pop_chunks()

    return waveform, metadata

waveform, metadata = load(SAMPLE_WAV_SPEECH_URL)
print(waveform.shape)
print(metadata)
```

```
torch.Size([95733105, 2])
StreamReaderSourceAudioStream(media_type='audio', codec='adpcm_ima_wav', codec_long_name='ADPCM IMA WAV', format='s16p', bit_rate=384000, num_frames=0, bits_per_sample=0, metadata={}, sample_rate=48000.0, num_channels=2)
```

<img width="1016" alt="Screen Shot 2022-11-27 at 4 38 13 PM" src="https://user-images.githubusercontent.com/855818/204160993-7ff68cad-c412-4304-8b0c-fefccfe909fe.png">

**Comment by mthrok** [comment]
We have migrate the default IO backend to FFmepg in nightly build. Going forward, installing torchaudio (nightly or future stable release) and FFmepg 5, should automatically workaround this issue.
